include/ddsrouter/event/impl/SignalHandler.ipp

```diff
@@ -131,20 +146,58 @@ void SignalHandler<SigNum>::signal_handler_routine_(
     }
 }
 
+template <int SigNum>
+void SignalHandler<SigNum>::signal_handler_thread_routine_() noexcept
+{
+    signal(SigNum, [](int)
+            {
+                signal_received_.store(true);
+                signal_received_cv_.notify_one();
+            });
+
+    signal_handler_active_.store(true);
+    while (true)
+    {
+        std::unique_lock<std::mutex> lock(signal_received_cv_mutex_);
+        signal_received_cv_.wait(
+            lock,
+            []
+            {
+                return signal_received_.load() || !signal_handler_active_.load();
+            });
+
+        if (signal_received_.load())
+        {
+            signal_received_.store(false);
```

Comment:
I would add a new `signal_handler_active_` if statement so it is explicit when to exit (just for code readability),